extensions/amp-truncate-text/1.0/truncate-text.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper} from '../../../src/preact/component';
+import {forwardRef} from '../../../src/preact/compat';
+import {truncateText} from './truncation';
+import {
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from '../../../src/preact';
+import {useStyles} from './truncate-text.jss';
+
+/**
+ * @param {!TruncateTextDef.Props} props
+ * @param {{current: (!TruncateTextDef.TruncateTextApi|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function TruncateTextWithRef(
+  {
+    persistentContent,
+    collapsedContent,
+    expandedContent,
+    onToggle,
+    children,
```

Comment:
Will we get a `children` key if it's not listed in the `AmpTruncateText['children']` configuration?